app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java

```diff
@@ -52,14 +47,12 @@ protected void onCreate(final Bundle savedInstanceState)
 
         ButterKnife.bind(this);
 
-        IBeaconScanner.getInstance().setCallback(this);
-
         // add input filters on {@link EditText}s.
         this.editTextMajor.setFilters(new InputFilter[]{new InputFilterMinMax(1, 65535)});
         this.editTextMinor.setFilters(new InputFilter[]{new InputFilterMinMax(1, 65535)});
 
         RxPermissions.getInstance(this)
-                .request(Manifest.permission.ACCESS_COARSE_LOCATION)
+                .request(Manifest.permission.ACCESS_FINE_LOCATION)
```

Comment:
Is fine location really necessary?

Reply:
@nielsdelporte ?

Reply:
See my reply from two days ago. I can't remember.

Reply:
Tested it myself, not necessary, will lower it again